### PR TITLE
ICU-20529 Generate res_index on partial resource trees.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20529
- [x] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by FrankYFTang** [comment]
I do not feel comfortable to review py file. Could you ask someone else to review it for you. All I can say is the change work when I try out complex rule in chrome/v8 ICU build

**Comment by sffc** [comment]
@markusicu The issue was that the SingleExecutionRequest for the res_index.res file was being deleted when a file filter was applied to the category because the SingleExecutionRequest had no input files listed, which is also the condition used to prune out unnecessary requests.  This change moves the SingleExecutionRequest into the flattening step, which happens after file filtering is executed.  Note that if the res_index.res file truly does need to be removed, this will still happen, because the IndexRequest does have the list of input files.

File: icu4c/source/data/buildtool/request_types.py [hunk 1]
```diff
@@ -303,8 +306,16 @@ def apply_file_filter(self, filter):
     def flatten(self, config, all_requests, common_vars):
         return PrintFileRequest(
             name = self.name,
-            output_file = self.output_file,
+            output_file = self.txt_file,
             content = self._generate_index_file(common_vars)
+        ).flatten(config, all_requests, common_vars) + SingleExecutionRequest(
```
Comment (markusicu) [inline icu4c/source/data/buildtool/request_types.py]: Please add a line feed after the plus sign.
I suggest indentation like this:
    return
        PrintFileRequest(
            keys = values
        ).flatten(args) +
        SingleExecutionRequest(
            keys = values
        ).flatten(args)

**Comment by sffc** [comment]
I rebased my branch to master, and then amended to address Markus's suggestion.

**Pull request merged**